### ISSUE TITLE
Add markerfacecolor & markeredgecolor params to _paint_carto_legend

### ISSUE THREAD

**Issue by HTenkanen** [ResidentMario/geoplot#35]
Hi! Awesome work!

I've been using geoplot to make pointplots such as [these](http://www.residentmar.io/geoplot/examples/nyc-collisions-map.html). 

![image](https://cloud.githubusercontent.com/assets/6659534/26354830/085947be-3fcf-11e7-9508-8c55597028bc.png)

My points are colored with a single color and hence I wanted that my legend would have the same markeredgecolor and facecolor that my points on the map had. I were exploring your codes and it seems that only way to change the legend coloring in _paint_carto_legend was to go and change the source code.

Hence, I would suggest to adding 'markerfacecolor' and 'markeredgecolor' parameters into the legend_kwargs which would allow to adjust those accordinly. 

Keep up the good work!
Henrikki

**Comment by Casyfill** [comment]
Hey there. Working on that in #34

**Comment by ResidentMario** [comment]
As of #114, `markerfacecolor` and `markeredgecolor` are now propagated to the legend when you pass `color`/`facecolor` and/or `edgecolor` parameters to your plot.

Sorry to have left this issue open for so long. There are a ton of improvements to `geoplot` coming out in the next 0.3.0 release, which fixes many many longstanding issues like this and others, you should check it out again once it's done. 😄

**Comment by ResidentMario** [comment]
Following up to say that this is now addressed more directly by #127.

On `geoplot@master` you can now pass `marker*` arguments to `legend_kwargs` (specifically: {`marker`, `markersize`, `markeredgecolor`, `markeredgewidth`, `markerfacecolor`}), and those parameters will be passed to the legend markers.

For example, here is a code snippet demonstrating the current defaults:

```python
import geopandas as gpd
import geoplot as gplt
import geoplot.crs as gcrs

continental_usa_cities = gpd.read_file(gplt.datasets.get_path('usa_cities'))
continental_usa_cities = continental_usa_cities.query('STATE not in ["AK", "HI", "PR"]')
contiguous_usa = gpd.read_file(gplt.datasets.get_path('contiguous_usa'))

ax = gplt.pointplot(
    continental_usa_cities, projection=gcrs.AlbersEqualArea(),
    scale='POP_2010', limits=(2, 30), hue='POP_2010',
    legend=True, legend_var='scale', legend_labels=range(1,6)
)
gplt.polyplot(contiguous_usa, ax=ax)
```

![image](https://i.imgur.com/GFEoPp3.png)

We can fiddle with the legend marker params to get something very different:

```python
import geopandas as gpd
import geoplot as gplt
import geoplot.crs as gcrs

continental_usa_cities = gpd.read_file(gplt.datasets.get_path('usa_cities'))
continental_usa_cities = continental_usa_cities.query('STATE not in ["AK", "HI", "PR"]')
contiguous_usa = gpd.read_file(gplt.datasets.get_path('contiguous_usa'))

ax = gplt.pointplot(
    continental_usa_cities, projection=gcrs.AlbersEqualArea(),
    scale='POP_2010', limits=(2, 30), hue='POP_2010',
    legend=True, legend_var='scale', legend_labels=range(1,6),
    legend_kwargs={
        'markeredgecolor': 'gray',
        'markeredgewidth': 3,
        'markerfacecolor': 'lightgray',
        'marker': '*'
    }
)
gplt.polyplot(contiguous_usa, ax=ax)
```

![](https://i.imgur.com/VHbEFzk.png)

Hope that helps. 😄
